src/mixins/object_origin.mixin.ts

```diff
@@ -1,345 +1,404 @@
+   * @param {TOriginY} originY Vertical origin: 'top', 'center' or 'bottom'
+   * @return {Point}
+   */
+  getPointByOrigin(originX: TOriginX, originY: TOriginY): Point {
+    return this.translateToOriginPoint(
+      this.getRelativeCenterPoint(),
+      originX,
+      originY
+    );
+  }
+
+  /**
+   * Returns the normalized point (rotated relative to center) in local coordinates
+   * @param {Point} point The point relative to instance coordinate system
+   * @param {TOriginX} originX Horizontal origin: 'left', 'center' or 'right'
+   * @param {TOriginY} originY Vertical origin: 'top', 'center' or 'bottom'
+   * @return {Point}
+   */
+  normalizePoint(point: Point, originX: TOriginX, originY: TOriginY): Point {
+    const center = this.getRelativeCenterPoint();
+    let p, p2;
+    if (typeof originX !== 'undefined' && typeof originY !== 'undefined') {
+      p = this.translateToGivenOrigin(
+        center,
+        'center',
+        'center',
+        originX,
+        originY
+      );
+    } else {
+      p = new Point(this.left, this.top);
+    }
+
+    if (this.angle) {
+      p2 = point.rotate(-degreesToRadians(this.angle), center);
+    } else {
+      p2 = point;
+    }
+    return p2.subtract(p);
+  }
+
+  /**
+   * Sets the position of the object taking into consideration the object's origin
+   * @param {Point} pos The new position of the object
+   * @param {TOriginX} originX Horizontal origin: 'left', 'center' or 'right'
+   * @param {TOriginY} originY Vertical origin: 'top', 'center' or 'bottom'
+   * @return {void}
+   */
+  setPositionByOrigin(pos: Point, originX: TOriginX, originY: TOriginY) {
+    const center = this.translateToCenterPoint(pos, originX, originY),
+      position = this.translateToOriginPoint(
+        center,
+        this.originX,
+        this.originY
+      );
+    this.set({ left: position.x, top: position.y });
+  }
+
+  /**
+   * Sets the origin/position of the object to it's center point
+   * @private
+   * @return {void}
+   */
+  _setOriginToCenter() {
+    this._originalOriginX = this.originX;
+    this._originalOriginY = this.originY;
+
+    const center = this.getRelativeCenterPoint();
+
+    this.originX = 'center';
+    this.originY = 'center';
+
+    this.left = center.x;
+    this.top = center.y;
+  }
+
+  /**
+   * Resets the origin/position of the object to it's original origin
+   * @private
+   * @return {void}
+   */
+  _resetOrigin() {
+    if (
+      this._originalOriginX !== undefined &&
+      this._originalOriginY !== undefined
+    ) {
+      const originPoint = this.translateToOriginPoint(
+        this.getRelativeCenterPoint(),
+        this._originalOriginX,
+        this._originalOriginY
+      );
+
+      this.left = originPoint.x;
+      this.top = originPoint.y;
+
+      this.originX = this._originalOriginX;
+      this.originY = this._originalOriginY;
+      this._originalOriginX = undefined;
+      this._originalOriginY = undefined;
+    }
```

Comment:
would love to refactor those to be just inside the object.rotate method, and remove those as standalone methods. Not today

Reply:
Agreed
I would add `normalizePoint` to the list.
I think it is used only by controls.